ci/vcpkg/x64-osx-static-debug.cmake

```diff
@@ -21,5 +21,6 @@ set(VCPKG_LIBRARY_LINKAGE static)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES x86_64)
+set(VCPKG_OSX_DEPLOYMENT_TARGET "10.13")
```

Comment:
[Here is a build error](https://github.com/ursacomputing/crossbow/runs/2898078973#step:7:138) passing the wrong isysroot despite that I hardcoded the `VCPKG_OSX_DEPLOYMENT_TARGET` variable.

Reply:
```text
  /Users/runner/work/crossbow/crossbow/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-02-24-d67989bce1043b98092ac45996a8230a059a2d7e/include/vcpkg/base/files.h:46:5: error: 'path' is unavailable: introduced in macOS 10.15
      path u8path(vcpkg::StringView s);
      ^
  /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/filesystem:738:24: note: 'path' has been explicitly marked unavailable here
  class _LIBCPP_TYPE_VIS path {
                         ^
```

It seems that we can't use "10.13" here with Xcode 12.4. Can we use "10.15" or later hre?

Reply:
We should provide wheels compatible with `10.13`. If the XCode version is an issue we may try to use an erlier Xcode https://github.com/marketplace/actions/setup-xcode-version

Reply:
Though according to https://developer.apple.com/support/xcode/ we should be able to use Xcode 12.4 with macOS SDK 10.13. 
Could we change the isysroot argument during the [vcpkg compilation](https://github.com/ursacomputing/crossbow/runs/2898078973#step:7:133)?

Reply:
Or you mean that we need to compile vcpkg itself with macOS SDK 10.15 then we can switch to 10.13 to install the dependencies?

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html
We can change the `-isystem` value by `CMAKE_OSX_SYSROOT` CMake variable or `SDKROOT` environment variable.

Reply:
Well, I've [hardcoded the SDKROOT variable too](https://github.com/apache/arrow/pull/10587/files#diff-e81a171228b21e6f895ee07303be13edb78f40129d481044a95a6b124e7c7c33R23), but seems like it doesn't have an effect.